nav2_controller/src/controller_server.cpp

```diff
@@ -298,7 +300,7 @@ ControllerServer::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
    */
   costmap_ros_->deactivate();
 
-  publishZeroVelocity();
+  onGoalExit();
```

Comment:
I think on deactivation, `publishZeroVelocity` is the right behavior, no? We don't want our robot still moving at all if this is being shut down

Reply:
I altered the PR; but maybe we should publish a zero velocity on the `on_deactivate` method of the velocity publisher.